internal/project/core_test.go

```diff
@@ -53,3 +56,121 @@ func TestCore_ExistsWithID(t *testing.T) {
 	assert.True(t, ok)
 	assert.NoError(t, err)
 }
+
+func TestCore_Get(t *testing.T) {
+	type fields struct {
+		repo IRepo
+	}
+	type args struct {
+		ctx       context.Context
+		projectID string
+	}
+	ctrl := gomock.NewController(t)
+	mockRepo := mocks.NewMockIRepo(ctrl)
+	project := getDummyProjectModel()
+	ctx := context.Background()
+
+	testModel := &Model{}
+	testModel.SetVersion("123")
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *Model
+		wantErr bool
+	}{
+		{
+			name: "Test1",
+			fields: fields{
+				repo: mockRepo,
+			},
+			args: args{
+				ctx:       ctx,
+				projectID: project.ProjectID,
+			},
+			want:    testModel,
+			wantErr: false,
+		},
+		{
+			name: "Test2",
+			fields: fields{
+				repo: mockRepo,
+			},
+			args: args{
+				ctx:       ctx,
+				projectID: "",
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Core{
+				repo: tt.fields.repo,
+			}
+			var err2 error = nil
+			if len(tt.args.projectID) == 0 {
+				err2 = fmt.Errorf("Invalid Project ID!")
+			}
+			mod := &Model{}
+			mockRepo.EXPECT().Get(gomock.Any(), gomock.Any(), mod).Do(func(arg1 context.Context, arg2 string, mod *Model) {
+				mod.SetVersion("123")
```

Comment:
In case of an invalid project ID, this will still set a version value for the Model. Please fix this.